src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderProgramProfileRepository.java

```diff
@@ -1,10 +1,25 @@
 package uk.gov.justice.hmpps.prison.repository.jpa.repository;
 
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderProgramProfile;
 
+import java.time.LocalDate;
 import java.util.List;
 
 public interface OffenderProgramProfileRepository extends CrudRepository<OffenderProgramProfile, Long> {
     List<OffenderProgramProfile> findByOffenderBooking_BookingIdAndProgramStatus(Long bookingId, String programStatus);
+
+    @Query(
```

Comment:
I cut this down to the minimum, but it is still too much JPQL to put in the method name